### PR TITLE
Test: Move analytics tests

### PR DESCRIPTION
**Pull request by blowery**
These were fun. They add in a fake global Image object, which is not provided by our fake dom. They then spy on how that object is used and assert various things.

The one really odd bit is how we track the loaded images. Instead of setting a new array, I reset the length to 0. This keeps the reference alive across all the various function contexts that are using it. Bit odd, but it works.

### CODE REVIEW

**Comment by blowery** [comment]
cc @gziolo

File: client/analytics/test/index.js [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
-	set: function( value ) {
-		this._src = value;
-		imagesLoaded.push( url.parse( value, true, true ) );
-	}
-} );
+	before( function spyOnImage() {
+		imagesLoaded.length = 0;
+		originalImage = global.Image;
+
+		global.Image = function() {
```
Comment (gziolo) [inline client/analytics/test/index.js]: Nice mock :D

File: client/analytics/test/lib/load-script/index.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 function fakeLoader( url, callback ) {
 	fakeLoader.urlsLoaded.push( url );
 	if ( callback ) {
-		setTimeout( function() { callback(); }, 0 );
+		setTimeout( function() {
```
Comment (gziolo) [inline client/analytics/test/lib/load-script/index.js]: It's minor, but we could also use `async` lib and call [async.nextTick](https://www.npmjs.com/package/async#nextTick).

**Comment by gziolo** [comment]
Looks good and works as expected. I tested it also in isolation using `.only` helper.

:shipit:

File: client/analytics/test/lib/load-script/index.js (hunk 2 above)
Comment (blowery) [inline client/analytics/test/lib/load-script/index.js]: I'd prefer `lodash/defer` but yeah. I'm trying to get us off `async`

Comment (gziolo) [inline client/analytics/test/lib/load-script/index.js]: That works for me, too :)

**Pull request merged**